### PR TITLE
Sync yarn.lock between baseDir and installDir

### LINKED ISSUE #151: yarn.lock is not version controlled.
The yarn lock is in the target directory. It should be version controlled. I suggest it should go in the project `baseDirectory`. For example, `client/baseDirectory`.

> In order to get consistent installs across machines, Yarn needs more information than the dependencies you configure in your package.json. Yarn needs to store exactly which versions of each dependency were installed.

https://yarnpkg.com/lang/en/docs/yarn-lock

### PR DESCRIPTION
**Pull request by fdietze**
I was not able to test this locally...

(https://gitter.im/scalacenter/scalajs-bundler?at=5b17a736edd06f7d151380b5)

fixes #151

### CODE REVIEW

**Comment by julienrf** [comment]
Is it possible to add scripted tests for this feature?

**Comment by fdietze** [comment]
Should be working correctly now! I tested it locally.

**Comment by fdietze** [comment]
Any feedback on this one? I'm using it locally with my team since one week, and it's working well.

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala [hunk 1]
```diff
@@ -33,20 +34,47 @@ object Npm extends ExternalCommand("npm")
...
     }
 
+  def install(baseDir:File, installDir: File, useYarn: Boolean, logger: Logger): Unit =
+    if (useYarn) {
+      syncYarnLockfile(baseDir, installDir, logger) {
+        Yarn.run("install" +: yarnOptions: _*)(installDir, logger)
+      }
+    } else {
+      Npm.run("install")(installDir, logger)
+    }
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala]: Why do we need to distinguish between `yarn add` and `yarn install`, whereas this was not the case before?

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala [hunk 2]
```diff
@@ -33,20 +34,47 @@ object Npm extends ExternalCommand("npm")
...
+    val targetLockFile = installDir / "yarn.lock"
+    if(sourceLockFile.exists()) {
+      logger.info("Using lockfile " + sourceLockFile)
+      IO.copyFile(sourceLockFile, targetLockFile)
+    }
+
+    yarnCommand
+
+    if(targetLockFile.exists()) {
+      logger.info("Wrote lockfile to " + sourceLockFile)
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala]: Maybe put that line to the `debug` level?

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala (hunk 1 above)
Comment (fdietze) [reply sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala]: Both, `yarn add` and `yarn install` were there before, I just moved them into `ExternalCommand`:
https://github.com/scalacenter/scalajs-bundler/blob/master/sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala#L47
https://github.com/scalacenter/scalajs-bundler/blob/9c3cd1b13817a471149d4552db5396c0b9849bb9/sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/NpmUpdateTasks.scala#L32

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala (hunk 2 above)
Comment (fdietze) [reply sbt-scalajs-bundler/src/main/scala/scalajsbundler/ExternalCommand.scala]: good idea, I'll change it.

**Comment by julienrf** [comment]
Thanks!

**Pull request merged**